src/libs/Tokenizer.js

```diff
@@ -0,0 +1,95 @@
+
+
+var theTokenizer;
+
+class Tokenizer {
+
+    constructor(program, literalsList) {
+        this.program = program;
+        this.literals = literalsList;
+        this.tokens = [];
+        this.currentToken = 0;
+        this.tokenize();
+    }
+
+    tokenize() {
+        let tokenizedProgram = this.program;
+        tokenizedProgram = tokenizedProgram.replace(/\n/g, "");
+
+        // extracts replacing them with _$_, stores them in an array
+        // to be added back in later.
+        var texts = [...tokenizedProgram.matchAll(/".*"/g)];
+        tokenizedProgram = tokenizedProgram.replace(/".*"/g, "_$_");
+
+        // prevents leading 'lat' in lat lon from combining with variable name token
+        tokenizedProgram = tokenizedProgram.replace(/( \d+)/g, function(m){
+            return "_" + m;
+        });
+        console.log(tokenizedProgram);
+
+        this.literals.forEach((s) => {
+            let regex = new RegExp(s, "g");
+            tokenizedProgram = tokenizedProgram.replace(regex, "_"+s+"_");
+            console.log(tokenizedProgram);
+        });
+        //TODO: Below will make lat and lon hard to distinguish?? may need comma between them??
+        tokenizedProgram = tokenizedProgram.replace(/ /g, "");
+
+        texts.forEach((text) => {
+            tokenizedProgram = tokenizedProgram.replace(/\$/, text);
```

Comment:
Related to first comment regarding array _texts_; will the code inside this for loop ever run?

Reply:
?? of course it will, its simply replacing the $'s that I used to mark where the "text" should go.

Reply:
Sorry, I missed that. You're right.